x-pack/plugins/monitoring/server/lib/details/get_metrics.ts

```diff
@@ -72,3 +72,73 @@ export async function getMetrics(
     return data;
   });
 }
+
+// for requests that use metrics-* datastream, currently only elasticsearch and kibana integrations
```

Comment:
this is the same as getMetrics but requires a `productType` (elasticsearch, kibana, logstash, etc).  I didn't want to update all code yet that doesn't need the datastream indices but uses getMetrics.  It also no longer accepts the indexPattern.  instead the `productType` is passed to `getNewSeries` which is passed to `fetchNewMetrics` and the index pattern is created inside there. `ccs` is also passed along for ccs indices.